ls1intum/Artemis
# `General`: Improve course registration and further reduce loading time of the first page visit

### Checklist
#### General


#### Client

### Motivation and Context



Reduce bundle size of the client application by using more lazy modules in Angular. Lazy load the markdown module.

### Description


### Steps for Testing

Prerequisites:
- 1 Student
- 1 Course with student registration enabled

1. Log in to Artemis
2. Click on the course registration button
3. Check That the courses are displayed correctly
4. Register for a course
5. Navigate back
6. Check that you have been successfully registered

BONUS: Check bundle size for build

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage






### Screenshots



<img width="1016" alt="Bildschirmfoto 2021-11-23 um 09 09 52" src="https://user-images.githubusercontent.com/51987021/142994035-dcf83510-6de9-49be-a458-6dfe6537a47b.png">
<img width="1011" alt="Bildschirmfoto 2021-11-23 um 09 10 00" src="https://user-images.githubusercontent.com/51987021/142994032-a9118956-8edf-4f1b-9b4d-54fba0ed0058.png">
<img width="1003" alt="Bildschirmfoto 2021-11-23 um 09 45 36" src="https://user-images.githubusercontent.com/51987021/142994030-1457afa3-de52-4567-8dc7-eebeaa3c9649.png">
<img width="542" alt="Bildschirmfoto 2021-11-22 um 19 44 49" src="https://user-images.githubusercontent.com/51987021/143000719-76697d22-21f7-4bdc-ac8a-0aabf5b0dcbc.png">
